collector/collector.go

```diff
@@ -187,9 +190,46 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
 // Collect implements the prometheus.Collector interface.
 func (c *collector) Collect(ch chan<- prometheus.Metric) {
 	wg := sync.WaitGroup{}
-	wg.Add(len(c.devices))
+
+	var realDevices []config.Device
 
 	for _, dev := range c.devices {
+		if dev.Srv != "" {
+			log.WithFields(log.Fields{
+				"SRV": dev.Srv,
+			}).Info("SRV configuration detected")
+
+			dnsMsg := new(dns.Msg)
+			dnsCli := new(dns.Client)
+
+			dnsMsg.RecursionDesired = true
+			dnsMsg.SetQuestion(dns.Fqdn(dev.Srv), dns.TypeSRV)
+			r, _, err := dnsCli.Exchange(dnsMsg, "1.1.1.1:53")
```

Comment:
Can the nameserver also be added into the Config YAML somewhere with the 1.1.1.1:53 as a `const` default if one isn't supplied? Thinking internal corporate network infrastructure and internal private DNS only type situations.

Reply:
Sure this is a good idea, I'll add it to the config.

Reply:
Please review my last commit with requested changes.